packages/backend-deployer/package.json

```diff
@@ -26,5 +26,8 @@
   "peerDependencies": {
     "aws-cdk": "^2.110.1",
     "typescript": "^5.0.0"
+  },
+  "devDependencies": {
+    "@aws-amplify/cli-core": "^0.3.0"
```

Comment:
I think it's acceptable, but dangerous idea. (see my reply to https://github.com/aws-amplify/amplify-backend/pull/947#discussion_r1471430705 )

Reply:
I'll remove it and mock the `packageManagerControllerFactory` in this PR

Reply:
you can just mock `type PackageManagerController` after all refactorings right ?